jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamInitSelectionPanel.kt

```diff
@@ -0,0 +1,163 @@
+    private lateinit var templateComboBox: ComboBox<SamProjectTemplate>
+    private lateinit var fragments: Wrapper
+
+    private val wizardFragments: Map<WizardFragment, JComponent>
+
+    init {
+        // Source all templates, find all the runtimes they support, then filter those by what the IDE supports
+        val supportedRuntimeGroups = RuntimeGroup.registeredRuntimeGroups()
+        SamProjectTemplate.supportedTemplates().asSequence()
+            .flatMap { it.supportedRuntimes().asSequence() }
+            .filter(runtimeFilter)
+            .filter { supportedRuntimeGroups.contains(find { runtimeGroup -> runtimeGroup.runtimes.contains(it) }) }
+            .distinct()
+            .sorted()
+            .forEach { runtimeComboBox.addItem(it) }
+
+        setupSamSelectionElements(samExecutableField, editSamExecutableButton, samLabel)
+
+        runtimeComboBox.addActionListener {
+            runtimeUpdate()
+            wizardUpdate()
+        }
+
+        templateComboBox.addActionListener { wizardUpdate() }
+        templateComboBox.renderer = SimpleListCellRenderer.create { label, value, _ ->
+            label.text = value?.displayName()
+            label.toolTipText = value?.description()
+        }
+
+        wizardFragments = wizardFragmentList.associateWith {
+            val panel = JPanel(BorderLayout())
+            val fragmentTitle = it.title()
+            if (fragmentTitle != null) {
+                panel.border = IdeBorderFactory.createTitledBorder(it.title(), false)
+            }
+            panel.add(it.component(), BorderLayout.CENTER)
+            panel
+        }
+
+        fragments.setContent(
+            panel {
+                wizardFragments.values.forEach {
+                    row() {
+                        it(grow)
+                    }
+                }
+            }
+        )
+
+        runtimeUpdate()
+        wizardUpdate()
+    }
+
+    fun setRuntime(runtime: Runtime) {
+        val itemCount = runtimeComboBox.itemCount
+        for (itemIndex in 0 until itemCount) {
+            if (runtimeComboBox.getItemAt(itemIndex) == runtime) {
+                runtimeComboBox.selectedItem = runtime
+                return
+            }
+        }
+    }
+
+    private fun runtimeUpdate() {
+        val selectedRuntime = runtimeComboBox.selectedItem as? Runtime
+        templateComboBox.removeAllItems()
+
+        // if selected runtimeComboBox is null, we're on an unsupported platform
+        if (selectedRuntime == null) {
+            return
+        }
+        SamProjectTemplate.supportedTemplates()
+            .filter { it.supportedRuntimes().contains(selectedRuntime) }
+            .forEach { templateComboBox.addItem(it) }
+    }
+
+    private fun wizardUpdate() {
+        val selectedRuntime = runtimeComboBox.selectedItem as? Runtime
+        val selectedTemplate = templateComboBox.selectedItem as? SamProjectTemplate
+        wizardFragments.forEach { (wizardFragment, jComponent) ->
```

Comment:
nit: are parens needed to destructure the argument here, or are they optional?

Reply:
Nope, not needed